analyzer/runtime/evm/erc721.go

```diff
@@ -62,6 +84,58 @@ func evmDownloadTokenERC721(ctx context.Context, logger *log.Logger, source node
 	return &tokenData, nil
 }
 
+func evmDownloadNFTERC721(ctx context.Context, logger *log.Logger, source nodeapi.RuntimeApiLite, ipfsClient ipfsclient.Client, round uint64, tokenEthAddr []byte, id *big.Int) (*EVMNFTData, error) {
+	var nftData EVMNFTData
+	supportsMetadata, err := detectInterface(ctx, logger, source, round, tokenEthAddr, ERC721MetadataInterfaceID)
+	if err != nil {
+		return nil, fmt.Errorf("checking ERC721Metadata interface: %w", err)
+	}
+	//nolint:nestif
```

Comment:
Nit: It _is_ a little hard to read. Can we do an early exit instead? `if !supportMetadata { return ... }`

Actually, looks like we could do early return on several more ifs below, and have this function be pretty much linear. That would help readability a fair bit.

Reply:
early exits are more suitable for "something went wrong, bail." this is a section where we progressively obtain more info if it does support a certain interface. not supporting that interface is fine, and we may theoretically do other things to get information. following these guidelines for when to use early exit and when to do something inside an `if` block helps make code more conveniently composable

Reply:
I'll break this up into other methods tho. would be good to early-exit the metadata part if calling tokenURI fails for example

Reply:
refactored